domain/src/DomainLoader.php

```diff
@@ -95,6 +95,11 @@ public function loadMultipleSorted($ids = NULL) {
    * {@inheritdoc}
    */
   public function loadByHostname($hostname) {
+    // Strip www. off the front?
+    $www = \Drupal::config('domain.settings')->get('www_prefix');
```

Comment:
There is code replication here, but since both are public functions, I think we need to trim in both places. Best alternative would be to move this code to a helper inside the loader, since that is required by the Negotiator.